beets/importer.py

```diff
@@ -675,10 +688,16 @@ def find_duplicates(self, lib):
 
         duplicates = []
         task_paths = {i.path for i in self.items if i}
-        duplicate_query = dbcore.AndQuery((
-            dbcore.MatchQuery('albumartist', artist),
-            dbcore.MatchQuery('album', album),
-        ))
+        keys = config['import']['duplicate_keys'].as_str().split()
+        info = self.chosen_info().copy()
```

Comment:
We could even make `chosen_info` itself call `copy()` or `dict()` to make its contract neater: i.e., it returns a temporary/disposable dict.